inception_reports/generate_reports_manager.py

```diff
@@ -391,34 +766,161 @@ def export_data(project_data, output_directory=None):
         project_data (dict): The data to be exported.
     """
     current_date = datetime.now().strftime("%Y_%m_%d")
-    directory_name = f'exported_data_{current_date}'
+    directory_name = f"exported_data_{current_date}"
 
     if output_directory is None:
         output_directory = os.path.join(os.getcwd(), directory_name)
     else:
         output_directory = os.path.join(output_directory, directory_name)
-    
+
     if not os.path.exists(output_directory):
         os.makedirs(output_directory)
 
     project_name = project_data["project_name"]
-    with open(f"{output_directory}/{project_name.split('.')[0]}_data_{current_date}.json", "w") as output_file:
+    with open(
+        f"{output_directory}/{project_name.split('.')[0]}_data_{current_date}.json", "w"
+    ) as output_file:
         json.dump(project_data, output_file, indent=4)
-    st.success(f"{project_name.split('.')[0]} documents status exported successfully ✅")
+    st.success(
+        f"{project_name.split('.')[0]} documents status exported successfully ✅"
+    )
+
+
+def plot_project_progress(project) -> None:
+    """
+    Generate a visual representation of project progress based on a DataFrame of log data.
+
+    This function takes a DataFrame containing log data and generates
+    visualizations to represent the progress of different documents. It calculates the
+    total time spent on each document, divides it into sessions based on a specified
+    threshold, and displays a pie chart showing the percentage of finished and remaining
+    documents, along with a bar chart showing the total time spent on finished documents
+    compared to the estimated time for remaining documents.
+
+    Parameters:
+        project (dict): A dict containing project information, namely the name, tags, annotations, and logs.
+
+    """
+
+    # df = project["logs"]
+    project_name = project["name"].strip(".zip")
+    project_tags = project["tags"]
+    project_annotations = project["annotations"]
+    project_documents = project["documents"]
+
+    doc_categories = {
+        "ANNOTATION_IN_PROGRESS": 0,
```

Comment:
Best create constants for strings that are commonly used instead of using string literals.